test/built-ins/ArrayBuffer/prototype/transferToFixedLength/new-length-excessive.js

```diff
@@ -0,0 +1,28 @@
+// Copyright (C) 2023 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-arraybuffer.prototype.transfertofixedlength
+description: >
+  Throws a RangeError the newLength value is too large to create a new
+  ArrayBuffer.
+info: |
+  ArrayBuffer.prototype.transferToFixedLength ( [ newLength ] )
+
+  1. Let O be the this value.
+  2. Perform ? RequireInternalSlot(O, [[ArrayBufferData]]).
+  3. If IsSharedArrayBuffer(O) is true, throw a TypeError exception.
+  4. If IsDetachedBuffer(O) is true, throw a TypeError exception.
+  5. If newLength is undefined, let newByteLength be
+     O.[[ArrayBufferByteLength]].
+  6. Else, let newByteLength be ? ToIntegerOrInfinity(newLength).
+  7. Let new be ? Construct(%ArrayBuffer%, « 𝔽(newByteLength) »).
+  [...]
```

Comment:
these aren't the spec steps for transferToFixedLength: https://tc39.es/proposal-arraybuffer-transfer/ only has two steps, and they call into https://tc39.es/proposal-arraybuffer-transfer/, and the byte length is ToIndex'd which clamps it - so it shouldn't be throwing, i think?

Reply:
(that suggests that the transfer tests also need updating)

Reply:
aha, thanks for clarifying, you're right

Reply:
I've removed the spec steps from both of those tests and changed the description to talk about clamping in ToIndex.